### DISCUSSION TITLE
ViewsLifeTime not available under manged properties

### DISCUSSION THREAD

**Discussion by IrCheema** [microsoft-search/pnp-modern-search#1443] [Q&A] [answered]
Hi,
I am trying to show the total number of views for modern news pagesin my search results. However ViewsLifeTime properties are not available. I tried to manually  add this property but it return null. However in  OOTB News web part one can display  number of views. Is their any way to get this fix in search results.
Does any body knows how to implement (ms graph toolkit) for [Get itemAnalytics](https://docs.microsoft.com/en-us/graph/api/itemanalytics-get?view=graph-rest-1.0&tabs=http) in Search results web part.
Thanks in advance.

**Comment by FredrikEkstroem** [comment, marked as answer]
Did you add ViewsLifeTime to both Layout Slots and Selected Properties on the first webpart settings panel? And then call it from the template like 
```javascript
{{slot item @root.slots.ViewsLifeTime}}
```
that works for me

**Reply by IrCheema** [reply]
Thanks Fredrik for your response. There is actually an issues that i am facing in the webpart as  ViewsLifeTime is no longer available in the list of managed properties. So as per your suggestion i added the property manually in the Layout Slots and Selected Properties. I updated one of the card fields to show views, but still no success.  In the debug mode i noticed that value for ViewsLifeTime is null for every result. While same results under OOTB news web part displaying  number of views.

![Modern Search V4](https://user-images.githubusercontent.com/22212050/138917298-5a90dfcc-bfc7-4f2b-8921-e9acd36acb6e.png)
![OOTB_News](https://user-images.githubusercontent.com/22212050/138917337-5d2309fe-dad5-41a2-8b67-a7fe4d173d7b.jpg)
![Modern Search V4_ViewsLifeTimw](https://user-images.githubusercontent.com/22212050/138917524-c7132aa0-d581-4ea6-9aa8-0251c3ba55cf.jpg)

**Reply by wobba** [reply]
Can you explain your screenshot on what L37-38 are? I think you are affected by a bug which is fixed in v4.4 where when you enter a prop it's stored as the comma separated string - not as individual props. So the request is then wrong.

**Reply by IrCheema** [reply]
Thanks @wobba & @FredrikEkstroem , after updating to version v4.4 and implementing the solution suggested by Fredrik  i am now able to get ViewsLifeTime for most of the news pages. While comparing with OOTB News web part i have noticed that there is a missmatch between number of views and for one of the news pages ViewsLifeTime is null. But i guess OOTB News web part is displaying views via itemAnalytics (Get itemAnalytics - graph api). 
@wobba : Is there any way we can use itemAnalytics instead of ViewsLifeTime?

**PnP Search V4 (4.4)**
![image](https://user-images.githubusercontent.com/22212050/139051841-1b333725-ee25-4e53-901f-d139c6068b69.png)

**News Web Part**
![image](https://user-images.githubusercontent.com/22212050/139053897-39a6dfd9-6b35-4036-b870-b585478402e1.png)
![image](https://user-images.githubusercontent.com/22212050/139053969-adcc35f5-6c04-42f6-9c19-f87c85b091a9.png)

BR

**Reply by wobba** [reply]
If you implement a custom web component to run the API call needed you can do what you want. Also look at the other "views*" properties and see if you find the one matching the count you see on a page itself. They should be the same, or close, as there is some data sync in place.
